pkg/xen-tools/initrd/init-initrd

```diff
@@ -133,9 +133,26 @@ mount -t overlay -o rw,relatime,lowerdir=/mnt/modules,upperdir=/mnt/rootfs/tmp/m
 echo "Executing /mount_disk.sh"
 /mount_disk.sh
 
-# Commence launch sequence
-source /mnt/environment
+# Check if the environment file can be successfully processed by the shell,
+# otherwise it will break the execution and make the init process fail,
+# crashing the shim VM
+# shellcheck disable=SC1091,SC2091
+if $(. /mnt/environment); then
+    # shellcheck disable=SC1091
+    . /mnt/environment
```

Comment:
Didn't you already execute the file in the if statement?

Reply:
no, the execution in the if is done in a sub-shell (new instance) since it's enclosed by $(...), that's why we don't crash if something is wrong... this sub-shell will not succeed and fall in the else condition... if everything is correct, the if will succeed and then we need to execute the source in the current instance...

Reply:
ok